File: test-suite-python/src/test/python/micronaut/docs/operator/OnUpdateFilter.py
# tag::reconciler[]
from java.util.function import BiPredicate
from jakarta.inject import Singleton

from io.kubernetes.client.openapi.models import V1ConfigMap


@Singleton
class OnUpdateFilter(BiPredicate[V1ConfigMap, V1ConfigMap]):

    def test(self, old_obj: V1ConfigMap, new_obj: V1ConfigMap) -> bool:
        annotations = new_obj.getMetadata().getAnnotations()
        if annotations is not None:
            return "io.micronaut.operator" in annotations
        return False
# end::reconciler[]

File: test-suite/src/test/java/io/micronaut/docs/operator/OnUpdateFilter.java
package io.micronaut.docs.operator;
//tag::reconciler[]
import io.kubernetes.client.openapi.models.V1ConfigMap;
import jakarta.inject.Singleton;

import java.util.function.BiPredicate;

@Singleton
public class OnUpdateFilter implements BiPredicate<V1ConfigMap, V1ConfigMap> {

    @Override
    public boolean test(V1ConfigMap oldObj, V1ConfigMap newObj) {
        if (newObj.getMetadata().getAnnotations() != null) {
            return newObj.getMetadata().getAnnotations().containsKey("io.micronaut.operator");
        }
        return false;
    }
}
//end::reconciler[]
